core/src/main/kotlin/org/evomaster/core/problem/rest/resource/RestResourceCalls.kt

```diff
@@ -41,29 +62,104 @@ class RestResourceCalls(
 
         copy.isDeletable = isDeletable
         copy.shouldBefore.addAll(shouldBefore)
+        copy.is2POST = is2POST
 
         return copy
     }
 
     /**
      * @return genes that represents this resource, i.e., longest action in this resource call
      */
-    fun seeGenes() : List<out Gene>{
-        return longestPath().seeGenes()
+    fun seeGenes(filter : GeneFilter = GeneFilter.NO_SQL) : List<out Gene>{
+        return when(filter){
+            GeneFilter.NO_SQL -> actions.flatMap(RestAction::seeGenes)
+            GeneFilter.ONLY_SQL -> seeMutableSQLGenes()
+            GeneFilter.ALL-> seeMutableSQLGenes().plus(actions.flatMap(RestAction::seeGenes))
+            else -> throw IllegalArgumentException("there is no initialization in an ResourceCall")
+        }
     }
 
-    fun repairGenesAfterMutation(gene: Gene? = null){
-        val target = longestPath()
-        if(gene != null) repairGenePerAction(gene, target)
-        else{
-            actions.filter { it is RestCallAction && it != target }
-                    .forEach{a-> (a as RestCallAction).bindToSamePathResolution(target as RestCallAction)}
+    fun seeActions() : List<out Action> = dbActions.plus(actions)
+
+    /**
+     * @return the mutable SQL genes and they do not bind with any of Rest Actions
+     *
+     * */
+    private fun seeMutableSQLGenes() : List<out Gene> = getResourceNode().getMutableSQLGenes(dbActions, getRestTemplate(), is2POST)
+
+    fun repairGenesAfterMutation(mutatedGene: MutatedGeneSpecification?, cluster: ResourceCluster){
+
+        mutatedGene?: log.warn("repair genes of resource call ({}) with null mutated genes", getResourceNode().getName())
+
+        val boundBaseGenes = if (mutatedGene?.didStructureMutation() == true) longestPath().seeGenes() else mutatedGene?.getMutated(true)?.mapNotNull { it.gene } ?: longestPath().seeGenes()
```

Comment:
the concept of dealing with bounded-genes should be independent of REST. so likely this code needs to be refactored out in a way that can be used in different problems. eg, maybe inside the mutator operator (which already does some post-processing if i remember correctly)

Reply:
also, didn't we discuss about having `BoundedGene` as a new gene?

Reply:
yes. this would be fixed with another branch.